tests/smoke_tests/test_basic.py

```diff
@@ -659,6 +659,68 @@ def test_kubernetes_context_failover():
         smoke_tests_utils.run_one_test(test)
 
 
+def test_launch_and_exec_async(generic_cloud: str):
+    """Test if the launch and exec commands work correctly with --async."""
+    name = smoke_tests_utils.get_cluster_name()
+    to_down = smoke_tests_utils.get_cluster_name() + '-down'
+    test = smoke_tests_utils.Test(
+        'launch_and_exec_async',
+        [
+            f'sky launch -c {name} -y --async',
+            # Async exec.
+            f'sky exec {name} echo --async',
```

Comment:
We should make sure `sky exec` without `--async` have reasonable hints.

Reply:
Currently it uses a spinner to indicate the progress:

```
> sky exec t1 echo
Command to run: echo
Submitting job to cluster: t1
⠋ Waiting for cluster t1 to be started.
```

But the spinner is cancelled after PENDING stage ends, will investigate how to validate this in smoke test